pkg/api/api.go

```diff
@@ -335,6 +335,7 @@ func (hs *HTTPServer) registerRoutes() {
 				dashboardRoute.Get("/uid/:uid/img/:kind/:theme", hs.ThumbService.GetImage)
 				dashboardRoute.Post("/uid/:uid/img/:kind/:theme", hs.ThumbService.SetImage)
 				dashboardRoute.Put("/uid/:uid/img/:kind/:theme", hs.ThumbService.UpdateThumbnailState)
+				dashboardRoute.Get("/previews/system-requirements", hs.ThumbService.GetSystemRequirements)
```

Comment:
@ryantxu this API will be used by regular users, not just admins. It looks like all the current APIs with `/api/admin` prefix are intended to be used only by admins - do you think its ok to change that?

<img width="733" alt="image" src="https://user-images.githubusercontent.com/23451458/153284468-55210f09-d360-4e59-a2c1-2d5edece63e6.png">

Reply:
:thinking: I am still pretty confused by this.

We currently only surface config errors to admins (if we do at all!!!)  it seems like previews should just not be available to the end user unless they are installed properly.

Essentially the end user should never see the feature toggle set to true unless it is actually configured properly -- right?

Reply:
rather than a new API call... perhaps we can/should add a general `setupError` flag section in frontend settings.  This would also let us avoid the extra redux store/provider

Reply:
> Essentially the end user should never see the feature toggle set to true unless it is actually configured properly -- right?

This message mimics the one presented in the `Share Panel` modal:

![image](https://user-images.githubusercontent.com/23451458/153308934-1a02e3f1-f58d-4cfa-9da4-b1e73f8fe849.png)


I think we discussed the for/against with @radiorental here https://raintank-corp.slack.com/archives/C01FJF3LESW/p1643306904035300?thread_ts=1643283822.029000&cid=C01FJF3LESW

Reply:
> rather than a new API call... perhaps we can/should add a general setupError flag section in frontend settings. This would also let us avoid the extra redux store/provider

I keep forgetting we can use the frontend settings! migrated to this structure at the moment

```
{
    systemRequirements: {
      met: boolean;
      requiredImageRendererPluginVersion: string;
    };
    thumbnailsExist: boolean;
  }
```

would have to think about how to migrate it to something more generic that fits all the scenarios